osu.Game/Rulesets/Scoring/ScoreProcessor.cs

```diff
@@ -99,7 +99,7 @@ public abstract class ScoreProcessor
         /// <summary>
         /// The default conditions for failing.
         /// </summary>
-        protected virtual bool DefaultFailCondition => Health.Value == Health.MinValue;
+        protected virtual bool DefaultFailCondition => Health.Value < Health.MinValue + 1e-15;
```

Comment:
I believe you can also use one of `Precision.AlmostEquals`, `Precision.AlmostBigger` or `Precision.DefinitelyBigger`.